src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_LABEL = MarketItemType.SUPPLY.toString().toLowerCase();
+    private static final String BUY_LABEL = MarketItemType.BUY.toString().toLowerCase();
+    private static final String RESULT_LABEL = "result";
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public String getStatistic(String fromFileName, String toFileName) {
+        MarketData marketData = new MarketData();
+
+        readFromFile(fromFileName, (line) -> {
+            String[] tokens = line.split(COMMA);
+            if (line.isEmpty() || tokens.length != 2) {
+                return;
+            }
+            String type = tokens[0].toLowerCase().trim();
+            int value = Integer.parseInt(tokens[1].trim());
+            if (type.equals(SUPPLY_LABEL)) {
+                marketData.addSupplyTotal(value);
+            } else if (type.equals(BUY_LABEL)) {
+                marketData.addBuyTotal(value);
+            }
+        });
+
+        String statistic = getStatisticString(marketData);
+        writeStatisticToFile(toFileName, statistic);
+
+        return statistic;
+    }
+
+    private void readFromFile(String fileName, LineProcessor processor) {
+        String line = "";
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while ((line = reader.readLine()) != null) {
+                processor.execute(line);
+            }
+        } catch (Exception e) {
+            String errStr = "Malformed line in " + fileName + ": '" + line + "'" + fileName;
```

readFromFile currently catches every Exception and rethrows with a confusing message that duplicates `fileName` and does not include a line number. Current code:
```
} catch (Exception e) {
    throw new RuntimeException("Malformed line in " + fileName + ": '" + line + "'" + fileName, e);
}
```
Problems and recommended fix:
- Distinguish I/O errors from data-processing (malformed-line) errors.
- For I/O errors rethrow as the checklist example recommends: `new RuntimeException("Can't read data from the file " + fileName, e)`.
- For data-processing errors (thrown from the line processor) ensure the thrown exception includes file name, line number and offending line and preserve the cause.

Quoted checklist guidance (example):
```
catch (Exception e) {
    throw new RuntimeException("Can't read data from the file " + fileName, e);
}
```
See "Don't ignore exceptions" in the checklist.